src/main/java/com/sparklicorn/bucket/games/tetris/TetrisGame.java

```diff
@@ -170,6 +171,25 @@ protected long calcPointsForClearing(int lines) {
 		return POINTS_BY_LINES_CLEARED.get(lines) * (level + 1L);
 	}
 
+	/**
+	 * Calculates the coordinates of blocks that make up the current shape,
+	 * in the given position.
+	 *
+	 * @param blockCoords Contains block coordinates that make up the shape.
+	 * The newly calculated block positions will be written here.
+	 * @param position The location and rotation of the shape to calculate block positions for.
+	 * @return The modified blockCoords array (for convenience).
+	 */
+	public Coord[] populateBlockPositions(Coord[] blockCoords, Position position) {
```

Comment:
On second thought, this probably didn't need to be moved from Shape. It makes sense to live in either class. Will just keep it here though.
This could be generalized by also passing in the Shape.